docs/reference/components.cue

```diff
@@ -391,6 +395,33 @@ components: {
 		metrics?: #MetricOutput
 	}
 
+	#IAM: {
```

Comment:
I didn't realize this until now. Do you think we're better off defining a `#AWSIAM` and `#GCPIAM` resource? I don't think they'll always perfectly align and would prefer to separate them.

In addition, we should probably structure the component documents to be something like:

```
permissions: aws_iam:
```

What do you think?

Reply:
Yeah, seems more future-proof to do it that. AWS and GCP are structurally very similar but that may not be true of platforms we support in the future. So I'm thinking maybe `#IAM` could be an enum instead.

Reply:
Hmmm. It's not clear to me that there's enough structural dissimilarity to warrant that. I'm open to reconsidering if we end up supporting a cloud platform that significantly departs from this mold, but for now I'm not sure I see the value of maintaining separate schemas when the only difference in actual output is how the action names and docs URLs are templatized.

Reply:
Ok. It's worth checking Azure also fits this mold, since we have Azure components that will need this as well.

Reply:
As far as I can tell, Azure is similar, with a resource naming scheme that's more URL-ish, e.g. `Microsoft.EventHub/clusters/operationresults/read`. But I don't really have much of a dog in this fight so if you want to go for the more future-proof enum path here, I'm cool with that.

Reply:
:+1: I'm fine with the current modeling. We can always change it later if they diverge.